numpy/core/fromnumeric.py

```diff
@@ -1019,6 +1019,41 @@ def argmin(a, axis=None, out=None):
     return _wrapfunc(a, 'argmin', axis=axis, out=out)
 
 
+def argmatch(a, v):
+    """
+    Find indices into `a` where elements in `v` match those in `a`.
+
+    Find the indices into an array `a` whose values match those queried in `v`.
+    Both arrays are first flattened to 1D, and the values in `a` must be unique.
+    This is an accelerated equivalent of:
+
+    `np.array([ int(np.where(a == val)[0]) for val in v ])`
+
+    Parameters
+    ----------
+    a : 1-D array_like
+        Input array.
+    v : array_like
+        Values to match against `a`.
+
+    Returns
+    -------
+    indices : array of ints
+        Array of indices into `a` with the same shape as `v`.
+
+    Adapted from http://stackoverflow.com/a/8251668
+    """
+    a, v = np.ravel(a), np.ravel(v)
+    if len(a) != len(np.unique(a)):
+        raise ValueError("values in `a` must be unique for unambiguous results")
```

Comment:
Seems to me that we should just allow this - computing `unique` is expensive, especially if the input is already unique